crates/linker/src/generate_dylib.rs

```diff
@@ -0,0 +1,421 @@
+use std::error::Error;
+
+use object::elf::{self, SectionHeader64};
+use object::read::elf::{FileHeader, ProgramHeader, SectionHeader, Sym};
+use object::Endianness;
+
+use object::write::elf::Writer;
+use roc_error_macros::internal_error;
+
+// an empty shared library, that we build on top of
+const DUMMY: &[u8] = include_bytes!("../dummy-elf64-x86-64.so");
+
+// index of the dynamic section
+const DYMAMIC_SECTION: usize = 4;
+
+pub fn generate(custom_names: &[String]) -> Result<Vec<u8>, Box<dyn Error>> {
```

Comment:
nit: could we technically use bumpalo vec throughout this?

Reply:
not easily. That buffer needs to be a `buffer: &'a mut dyn WritableBuffer` and that is of course not implemented for bumpalo vectors. We could wrap the bumpalo vector and implement it, but I don't think that's worth it.

Reply:
Ah. makes sense.